src/Orleans.Core/Streams/PubSub/StreamPubSubImpl.cs

```diff
@@ -75,7 +75,14 @@ public Task<int> ConsumerCount(Guid streamId, string streamProvider, string stre
             }
             else
             {
-                var implicitSubs = await implicitPubSub.GetAllSubscriptions(streamId);
+                List<StreamSubscription> implicitSubs = new List<StreamSubscription>();
+                //implicit pubsub requires a non-empty nor null namespace, so if the namespace is null or empty, there won't be any 
+                // subscription for that stream in implicitPubSub. Skip it to avoid unnecessary exception thrown
+                if (implicitPubSub.IsImplicitSubscribeEligibleStreamNameSpace(streamId))
```

Comment:
Implicit subscriptions are the domain of the implicitPubSub. it seems odd for the caller of GetAllSubscriptions to check this rather than for the GetAllSubscriptions to simply return no subscriptions if a namespace is invalid, similar to how RegisterProducer does.